compiler/src/dmd/dinterpret.d

```diff
@@ -6703,8 +6703,8 @@ ThrownExceptionExp chainExceptions(ThrownExceptionExp oldest, ThrownExceptionExp
     }
     // Little sanity check to make sure it's really a Throwable
     ClassReferenceExp boss = oldest.thrown;
-    const next = 5;                         // index of Throwable.next
-    assert((*boss.value.elements)[next].type.ty == Tclass); // Throwable.next
+    const next = 5;                         // index of Throwable._nextInChainPtr
+    assert((*boss.value.elements)[next].type.ty == Tpointer); // Throwable._nextInChainPtr
```

Comment:
Oh sorry, it can now apparently be both - `Tclass` or `Tpointer` (verified locally). I guess `Tpointer` if default-initialized to `null`, and `Tclass` if there's a chained exception. We're not checking the `Throwable` type definition as I thought, but the type of the CTFE value for the 6th field.

Reply:
OK, I did assert for either Tpointer or Tclass. Hopefully this is what you meant.